src/main/java/net/glowstone/net/handler/play/player/UseItemHandler.java

```diff
@@ -15,17 +17,22 @@ public void handle(GlowSession session, UseItemMessage message) {
         GlowPlayer player = session.getPlayer();
         ItemStack holding = player.getItemInHand();
 
-        if (holding != null) {
+        if (!InventoryUtil.isEmpty(holding)) {
             ItemType type = ItemTable.instance().getItem(holding.getType());
             if (type != null) {
-                if (type.canOnlyUseSelf()) {
+                if (type.getContext() == Context.AIR || type.getContext() == Context.AIR_BLOCK) {
                     type.rightClickAir(player, holding);
                 } else {
                     if (holding.getType() == Material.WATER_BUCKET || holding.getType() == Material.LAVA_BUCKET) {
                         holding.setType(Material.BUCKET);
                     }
                 }
             }
+
+            if (!InventoryUtil.isEmpty(holding) && holding.getAmount() <= 0) {
```

Comment:
That check doesn't make much sense, if the amount is `<= 0` then `isEmpty` will return true, thus breaking the condition.

[This](https://github.com/GlowstoneMC/Glowstone/blob/d4c3c57ce6bf9587563aad64b7d53b1bc30df56d/src/main/java/net/glowstone/entity/passive/GlowCow.java#L35-L40) is how we usually do it.